text/0000-bundle-groups.md

```diff
@@ -0,0 +1,89 @@
+
+# Guide-level explanation
+
+### Migration
+
+For users coming from Bundler 1, the `groups` command replaces the old options `--with` and `--without`, which allowed you to enable or disable groups on the fly while running `install`. In Bundler 1.99, the `--with` and `--without` options will print a deprecation warning when used, directing users to switch to using the `groups` command instead.
+
+### New `groups` command
+
+The `bundle groups` command allows you to list groups of gems declared in the Gemfile, and find out whether they will be installed or not. It also lets you enable or disable groups for future installs.
+
+Run the command `bundle groups` to list the groups defined in your Gemfile and find out whether they are enabled or disabled. Enabled groups will be installed when you run `bundle install`, and disabled groups will not be installed. Groups will be enabled by default, but they can be disabled by default by setting `optional: true` inside the Gemfile.
+
+Let’s say we have this example Gemfile:
+
+'' $ cat Gemfile
+'' gem "rails"
+'' group :production, optional: true do
+''   gem "sprockets-redirect"
+'' end
+'' group :test do
+''   gem "webmock"
+'' end
+
+In this Gemfile, the `rails` gem is in the default group, the gem `sprockets-redirect` is in the production group (which is disabled by default), and the gem `webmock` is in the test group. If you were to run `bundle groups`, this is what you would see:
+
+'' $ bundle groups
+''   - production   disabled  (default disabled)
+''   - test         enabled   (default enabled)
+
+Turning groups on is done via `bundle groups enable` and `bundle groups disable`. Here’s what it would look like to turn off the test group:
+
+'' $ bundle groups enable test
+'' Group `test` is now enabled
+
+After disabling the test group, running `bundle install` will install the gem `rails`, but will not install the gems `sprockets-redirect` or `webmock`.
+
+The `bundle groups` command exists to make the process of creating and installing groups understandable and predictable—you’ll always know what groups are going to be installed, and why those groups are enabled or disabled.
+
+### New `--only` option
+
+There are two ways to choose which groups will be installed. The first option simply uses the `groups enable` and `groups disable` commands to set your groups to be enabled and disabled as you want them to be.
+
+Alternatively, you can use the `--only` option (provided on both the `install` and `deploy` commands) to install only the groups whose names you give at that time. For example, if you were to run `bundle install --only production`, the `test` group would not be installed even though it is enabled by default, and the `production` group would be installed even though it is disabled by default.
```

Comment:
If I understand correctly, this is not an option for `groups` command but for `install` and `deploy` commands? This allows you to override the group enable/disable setting in the Gemfile or settings when running the command but it's unclear to me whether it will change the Gemfile/settings. 

I'm inclining that it should not change and just override for that one time use as a quick way to test things without having to change it back (I can be convinced otherwise if there's a good use case).

Reply:
The `--only` option for `install` already exists today. This paragraph is just describing how it would continue to work if we were to add a `groups` command. All flags are saved to settings in Bundler 1.x, and no flags are saved to settings in Bundler 2.x.

Reply:
Whoops! My bad. There was a pull request for an `only` option and we wound up never quite merging it. The `--only` option is a possible addition to allow installing and choosing groups at the same time with one command.